### PR TITLE
Volcano adapts to the k8s v1.25

### PR DESCRIPTION
**Pull request by wangyang0616**
Signed-off-by: wangyang <wangyang289@huawei.com>

/kind feature [#2510](https://github.com/volcano-sh/volcano/issues/2510)

1. Volcano adapts to the k8s v1.25
2. After Volcano is upgraded to K8S V1.25, the volumebinding package is imported to Volcano to ensure that csiStorageCapacity is compatible with K8S v1.23, v1.21, and v1.19. Use the csiStorageCapacity API of the v1beta1 version in the volumebinding package and do not upgrade the v1 version.

### CODE REVIEW

**Review by Thor-wl — COMMENTED** [review]
Is it nessary to update the volcano.sh/apis with k8s v1.25 first?

File: pkg/scheduler/plugins/volumebinding/volume_binding.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 	"fmt"
 	"sync"
 	"time"
+	"volcano.sh/volcano/cmd/scheduler/app/options"
```
Comment (jinzhejz) [inline pkg/scheduler/plugins/volumebinding/volume_binding.go:25]: suggest to move package `volcano.sh/volcano/cmd/scheduler/app/options` behind `k8s.io`

File: pkg/scheduler/plugins/volumebinding/binder.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ import (
 
 	v1 "k8s.io/api/core/v1"
 	storagev1 "k8s.io/api/storage/v1"
+	storagev1beta1 "k8s.io/api/storage/v1beta1"
```
Comment (jiangkaihua) [inline pkg/scheduler/plugins/volumebinding/binder.go:28]: Both `v1` and `v1beta1` is reserved in references, would they disturbe each other?

File: pkg/scheduler/plugins/volumebinding/binder.go [hunk 3]
```diff
@@ -227,7 +231,7 @@ func NewVolumeBinder(
 	pvcInformer coreinformers.PersistentVolumeClaimInformer,
 	pvInformer coreinformers.PersistentVolumeInformer,
 	storageClassInformer storageinformers.StorageClassInformer,
-	capacityCheck CapacityCheck,
+	capacityCheck *CapacityCheck,
```
Comment (jiangkaihua) [inline pkg/scheduler/plugins/volumebinding/binder.go:234]: Why does `capacityCheck` is modified to pointer?

File: cmd/controller-manager/app/server.go [hunk 4]
```diff
@@ -87,7 +87,7 @@ func Run(opt *options.ServerOption) error {
 	// add a uniquifier so that two processes on the same host don't accidentally both become active
 	id := hostname + "_" + string(uuid.NewUUID())
 
-	rl, err := resourcelock.New(resourcelock.ConfigMapsResourceLock,
+	rl, err := resourcelock.New(resourcelock.ConfigMapsLeasesResourceLock,
```
Comment (jiangkaihua) [inline cmd/controller-manager/app/server.go:90]: Why locktype is changed from `configmaps` to `configmapsleases`?

File: pkg/scheduler/plugins/predicates/predicates.go [hunk 5]
```diff
@@ -311,10 +311,10 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 	tolerationFilter := plugin.(*tainttoleration.TaintToleration)
 	// 5. InterPodAffinity
 	plArgs := &config.InterPodAffinityArgs{}
-	features := feature.Features{}
-	plugin, _ = interpodaffinity.New(plArgs, handle, features)
+	plugin, _ = interpodaffinity.New(plArgs, handle)
 	podAffinityFilter := plugin.(*interpodaffinity.InterPodAffinity)
 	// 6. NodeVolumeLimits
+	features := feature.Features{}
```
Comment (william-wang) [inline pkg/scheduler/plugins/predicates/predicates.go:317]: Does feature struct contain a lot of sub-feature?

File: pkg/scheduler/plugins/predicates/predicates.go [hunk 6]
```diff
@@ -398,7 +398,8 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 		}
 
 		// InterPodAffinity Predicate
-		status = podAffinityFilter.PreFilter(context.TODO(), state, task.Pod)
+		// TODO use framework.PreFilterResult
```
Comment (william-wang) [inline pkg/scheduler/plugins/predicates/predicates.go:401]: Do you want to add some thing more here? if not, please delete useless codes or comments.

File: pkg/scheduler/plugins/predicates/predicates.go [hunk 7]
```diff
@@ -415,7 +416,8 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 		}
 
 		// Check VolumeBinding: handle immediate claims unbounded case
-		status = volumebindingFilter.PreFilter(context.TODO(), state, task.Pod)
+		// TODO use framework.PreFilterResult
```
Comment (william-wang) [inline pkg/scheduler/plugins/predicates/predicates.go:419]: The same with above

File: pkg/scheduler/plugins/predicates/predicates.go [hunk 8]
```diff
@@ -433,7 +435,8 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 		}
 
 		// Check PodTopologySpread
-		status = podTopologySpreadFilter.PreFilter(context.TODO(), state, task.Pod)
+		// TODO use framework.PreFilterResult
```
Comment (william-wang) [inline pkg/scheduler/plugins/predicates/predicates.go:438]: The save as above

File: pkg/scheduler/cache/cache.go [hunk 9]
```diff
@@ -50,7 +50,7 @@ import (
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/klog"
 	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
-	volumescheduling "k8s.io/kubernetes/pkg/scheduler/framework/plugins/volumebinding"
+	volumescheduling "volcano.sh/volcano/pkg/scheduler/plugins/volumebinding"
```
Comment (william-wang) [inline pkg/scheduler/cache/cache.go:53]: cache should not depend plugin

Comment (william-wang) [reply pkg/scheduler/cache/cache.go:53]: suggest add capabilities directory like pkg/scheduler/capabilities/volumebinding and put our volumebinding there

File: pkg/scheduler/plugins/nodeorder/nodeorder.go [hunk 10]
```diff
@@ -212,12 +212,11 @@ func (pp *nodeOrderPlugin) OnSessionOpen(ssn *framework.Session) {
 	})
 
 	fts := feature.Features{
-		EnablePodAffinityNamespaceSelector: utilFeature.DefaultFeatureGate.Enabled(features.PodAffinityNamespaceSelector),
-		EnablePodDisruptionBudget:          utilFeature.DefaultFeatureGate.Enabled(features.PodDisruptionBudget),
-		EnablePodOverhead:                  utilFeature.DefaultFeatureGate.Enabled(features.PodOverhead),
-		EnableReadWriteOncePod:             utilFeature.DefaultFeatureGate.Enabled(features.ReadWriteOncePod),
-		EnableVolumeCapacityPriority:       utilFeature.DefaultFeatureGate.Enabled(features.VolumeCapacityPriority),
-		EnableCSIStorageCapacity:           utilFeature.DefaultFeatureGate.Enabled(features.CSIStorageCapacity),
+		EnableReadWriteOncePod:                       utilFeature.DefaultFeatureGate.Enabled(features.ReadWriteOncePod),
```
Comment (william-wang) [inline pkg/scheduler/plugins/nodeorder/nodeorder.go:215]: EnablePodDisruptionBudget is removed?

File: pkg/scheduler/plugins/nodeorder/nodeorder.go [hunk 11]
```diff
@@ -362,7 +361,7 @@ func (pp *nodeOrderPlugin) OnSessionOpen(ssn *framework.Session) {
 	ssn.AddNodeOrderFn(pp.Name(), nodeOrderFn)
 
 	plArgs := &config.InterPodAffinityArgs{}
-	p, _ = interpodaffinity.New(plArgs, handle, fts)
+	p, _ = interpodaffinity.New(plArgs, handle)
```
Comment (william-wang) [inline pkg/scheduler/plugins/nodeorder/nodeorder.go:364]: what's the reason of deleting fts for interpodaffinity?

File: pkg/scheduler/plugins/nodeorder/nodeorder.go [hunk 12]
```diff
@@ -371,7 +370,7 @@ func (pp *nodeOrderPlugin) OnSessionOpen(ssn *framework.Session) {
 	ptsArgs := &config.PodTopologySpreadArgs{
 		DefaultingType: config.SystemDefaulting,
 	}
-	p, _ = podtopologyspread.New(ptsArgs, handle)
+	p, _ = podtopologyspread.New(ptsArgs, handle, fts)
```
Comment (william-wang) [inline pkg/scheduler/plugins/nodeorder/nodeorder.go:373]: what's the reason of adding fts to podtopologyspread?

**Review by william-wang — APPROVED** [review]
/lgtm

**Pull request merged**